modules/rostests/apitests/ntdll/NtQueryInformationToken.c

```diff
@@ -0,0 +1,622 @@
+/*
+ * PROJECT:         ReactOS API tests
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:         Tests for the NtQueryInformationToken API
+ * COPYRIGHT:       Copyright 2022 George Bișoc <george.bisoc@reactos.org>
+ */
+
+#include "precomp.h"
+
+static
+HANDLE
+OpenCurrentToken(VOID)
+{
+    BOOL Success;
+    HANDLE Token;
+
+    Success = OpenProcessToken(GetCurrentProcess(),
+                               TOKEN_READ | TOKEN_QUERY_SOURCE | TOKEN_DUPLICATE,
+                               &Token);
+    if (!Success)
+    {
+        skip("OpenProcessToken() has failed to get the process' token (error code: %lu)!\n", GetLastError());
+        return NULL;
+    }
+
+    return Token;
+}
+
+static
+VOID
+QueryTokenUserTests(
+    _In_ HANDLE Token)
+{
+    NTSTATUS Status;
+    PTOKEN_USER UserToken;
+    ULONG BufferLength;
+/*
+ * FIXME: Make this stuff compile only on MSVC or Clang
+ * because apparently GCC doesn't understand what "%wZ"
+ * conversion type is on trace().
+ */
+#if defined(_MSC_VER) || (__clang__)
+    UNICODE_STRING SidString;
+#endif
+
+    Status = NtQueryInformationToken(Token,
+                                     TokenUser,
+                                     NULL,
+                                     0,
+                                     &BufferLength);
+    ok_hex(Status, STATUS_BUFFER_TOO_SMALL);
+
+    UserToken = RtlAllocateHeap(RtlGetProcessHeap(), 0, BufferLength);
+    if (!UserToken)
+    {
+        skip("Failed to allocate heap for token user!\n");
+        return;
+    }
+
+    Status = NtQueryInformationToken(Token,
+                                     TokenUser,
+                                     UserToken,
+                                     BufferLength,
+                                     &BufferLength);
+    ok_hex(Status, STATUS_SUCCESS);
+
+/* FIXME: See the comment above */
+#if defined(_MSC_VER) || (__clang__)
+    RtlConvertSidToUnicodeString(&SidString, UserToken->User.Sid, TRUE);
+    trace("=============== TokenUser ===============\n");
+    trace("The SID of current token user is: %wZ\n", &SidString);
+    trace("=========================================\n\n");
+    RtlFreeUnicodeString(&SidString);
+#endif
+
+    RtlFreeHeap(RtlGetProcessHeap(), 0, UserToken);
+}
+
+static
+VOID
+QueryTokenGroupsTests(
+    _In_ HANDLE Token)
+{
+    NTSTATUS Status;
+    PTOKEN_GROUPS Groups;
+    ULONG BufferLength;
+
+    Status = NtQueryInformationToken(Token,
+                                     TokenGroups,
+                                     NULL,
+                                     0,
+                                     &BufferLength);
+    ok_hex(Status, STATUS_BUFFER_TOO_SMALL);
+
+    Groups = RtlAllocateHeap(RtlGetProcessHeap(), 0, BufferLength);
+    if (!Groups)
+    {
+        skip("Failed to allocate heap for token groups!\n");
```

Comment:
Ditto, would make this a fail and include returned size.
(Not going to repeat it after this)